packages/projects-docs/pages/learn/integrations/tailscale.mdx

```diff
@@ -0,0 +1,116 @@
+---
+title: Tailscale
+description: Tailscale VPN integration that allows access to private resources from your Repositories.
+---
+
+import Callout from 'nextra-theme-docs/callout'
+
+# Tailscale
+
+At CodeSandbox, we use [Tailscale](https://tailscale.com/) to manage network access for private resources, including some of the services that make up CodeSandbox.
+As you work on an application in CodeSandbox Repositories, you might want to grant it access to private resources on your network.
+Tailscale can help.
+
+<Callout emoji="⭑">
+You can try Tailscale for free by following their [Quickstart guide](https://tailscale.com/kb/1017/install/#step-2-add-a-machine-to-your-network).
+Throughout this documentation, we'll assume you have a tailnet called `example-tailnet.ts.net` (which you should replace with your [name](https://tailscale.com/kb/1217/tailnet-name/)).
+</Callout>
+
+## Motivation
+
+Many applications exist as part of a collection of services.
+While things like mocks can help isolate a single application during development, this isn't always possible.
+Just because you develop in CodeSandbox, however, doesn't mean you can't connect your application to private services running in your organization's network.
+
+Integrating with Tailscale provides containers running in a CodeSandbox Repository environment with access to your private resources.
+Combined with Tailscale's Magic DNS and ACLs, you can develop your application as if it were hosted in the same network as your other services — without compromising security.
```

Comment:
Also consider linking to https://tailscale.com/kb/1081/magicdns/ and https://tailscale.com/kb/1018/acls/
```suggestion
Combined with Tailscale's MagicDNS and ACLs, you can develop your application as if it were hosted in the same network as your other services — without compromising security.
```